odoo/addons/base/models/res_currency.py

```diff
@@ -247,6 +247,29 @@ class CurrencyRate(models.Model):
         ('currency_rate_check', 'CHECK (rate>0)', 'The currency rate must be strictly positive.'),
     ]
 
+    @api.onchange('rate')
+    def _onchange_rate_warning(self):
+        if not self.currency_id.id:
+            return
+
+        latest_rate = self.search([
+            ('currency_id', '=', self.currency_id.id),
+            ('company_id', '=', self.company_id.id or self.env.company.id),
+            ('name', '<=', self.name or fields.Date.today()),
+        ], order="name desc", limit=1)
+        if latest_rate:
+            diff = (latest_rate.rate - self.rate) / latest_rate.rate
```

Comment:
`latest_rate.rate` could be 0, no?

Reply:
There is a SQL constraint to ensure it's greater than 0, but an additional check wouldn't cost anything :+1: